### PR TITLE
FRW-8639 Added OpenTelemetry integrator script.

### PR DESCRIPTION
**Pull request by kraal-spryker**
## PR Description
Ticket: https://spryker.atlassian.net/browse/FRW-8639

TBD

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

**Comment by kraal-spryker** [comment]
i see that you removed composer.json related changes. is it not needed anymore?

**Comment by gechetspr** [comment]
> i see that you removed composer.json related changes. is it not needed anymore?

We don't need any files to be downloaded first. Currently all installed via `composer require`

File: README.md [hunk 1]
```diff
@@ -1 +1,12 @@
-# opentelemetry-installer
\ No newline at end of file
+# opentelemetry-installer
+
+This is a simple script to integrate OpenTelemetry in the project. The script `otel_integrator.sh` should be copied to the root of the project.
```
Comment (annakotiuk) [inline README.md:3]: How do you like AI improvement?

This guide explains how to use the otel_integrator.sh script to integrate OpenTelemetry into your project.

### Script Overview

The otel_integrator.sh script automates the integration of OpenTelemetry into a project. It should be copied to the root of your project for proper functionality.

Parameters

**- Project Namespace (optional):**
  - Default: Pyz
  - Specifies the namespace of the project.
**- Deploy File Name (optional):**
  - Default: deploy.yml
  - Specifies the name of the deploy configuration file.
**- Install Script Name (optional):**
  - Default: config/install/docker.yml
  - Specifies the path to the installation script.

### Prerequisites
yq Tool:
Ensure the yq tool is installed on your system.
Example installation via Homebrew:
brew install yq

### Execution Steps

Copy the otel_integrator.sh script to the root directory of your project:
cp otel_integrator.sh /path/to/your/project/root
Run the script with the required parameters (or use defaults):
./otel_integrator.sh [PROJECT_NAMESPACE] [DEPLOY_FILE_NAME] [INSTALL_SCRIPT_NAME]
Example:

./otel_integrator.sh Pyz deploy.yml config/install/docker.yml
The script will modify the specified files as needed to integrate OpenTelemetry.

### Post-Script Review

After running the script:

**Verify the Deploy File:**
Review changes in the deploy file (deploy.yml by default) to ensure they align with your expectations.
**Run the Docker SDK:**
Start the Docker SDK using the updated deploy file:
docker/sdk boot [YOUR_DEPLOY_FILE_NAME]
**Validate Extensions:**
Confirm that the blackfire and opentelemetry extensions are not enabled simultaneously.

### Notes

Modify default values in the script if your project has specific requirements.
For troubleshooting, refer to logs or console output from the script execution.

Comment (annakotiuk) [inline README.md:3]: Or just fix errors: 
This is a simple script to integrate OpenTelemetry into the project

File: README.md [hunk 2]
```diff
@@ -1 +1,12 @@
...
+First param is a project namespace. Default to `Pyz`.
```
Comment (annakotiuk) [inline README.md:4]: The first parameter is the project namespace, which defaults to Pyz.

File: README.md [hunk 3]
```diff
@@ -1 +1,12 @@
...
+Second param is deploy file name. Default to `deploy.yml`
+Third param is install script name. Default to `config/install/docker.yml`
```
Comment (annakotiuk) [inline README.md:6]: The third parameter is the install script name, which defaults to config/install/docker.yml.

File: README.md [hunk 4]
```diff
@@ -1 +1,12 @@
...
+
+## Prerequisites
+Tool `yq` should be installed in the system. E.g. via brew `brew install yq`
```
Comment (annakotiuk) [inline README.md:9]: The yq tool must be installed `on` the system

File: README.md [hunk 5]
```diff
@@ -1 +1,12 @@
...
+
+## Review
+After the script is done, review your changes. Make sure that deploy file is updated and run `docker/sdk boot {%YOUR_FILE_NAME%}`. Make sure that `blackfire` extension and `opentelemetry` extension are not enabled in the same time.
```
[the remark this answers is not in the document]
Comment (annakotiuk) [reply README.md:12]: `at` the same time.

File: README.md [hunk 6]
```diff
@@ -1 +1,61 @@
...
+- --image-file:
+    - Default: same as --deploy-file
+    - In cases when your PHP image information is located in the different file, you can specify it via this param. If no value provided, script assumes that deploy file has all required fields.
+
+- --install-file:
+    - Default: config/install/docker.yml
+    - Specifies the install configuration file to update. This is needed in order to re-generate hook files on each install/deploy.
+
+- --base-image:
+    - Default: volhovm/spryker-8.3-alpine-3.20
```
Comment (olhalivitchuk) [inline README.md:26]: Aren't we going to put the image in spryker repo?

**Review by olhalivitchuk — CHANGES_REQUESTED** [review]

**Review by olhalivitchuk — APPROVED** [review]

**Review by asmarovydlo — APPROVED** [review]

**Pull request merged**